From ebe6e465c43321571fdd0d38f99b69e8eebdc7bc Mon Sep 17 00:00:00 2001
From: Elaine Zhang <zhangqing@rock-chips.com>
Date: Mon, 13 Jul 2020 15:43:27 +0800
Subject: [PATCH] dt-bindings: can: rockchip_canfd: add canfd support for
 rockchip

This patch add the rockchip canfd to the device tree bindings
documentation.

Change-Id: Ief6ffeeda87e28e8b70a706c92eba526e0db0686
Signed-off-by: Elaine Zhang <zhangqing@rock-chips.com>
---
 .../bindings/net/can/rockchip_canfd.txt       | 32 +++++++++++++++++++
 1 file changed, 32 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/can/rockchip_canfd.txt

diff --git a/Documentation/devicetree/bindings/net/can/rockchip_canfd.txt b/Documentation/devicetree/bindings/net/can/rockchip_canfd.txt
new file mode 100644
index 000000000000..b649b8b43c33
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/can/rockchip_canfd.txt
@@ -0,0 +1,32 @@
+Rockchip CANFD controller Device Tree Bindings
+---------------------------------------------------------
+
+Required properties:
+- compatible		: Should be:
+			  - "rockchip,canfd-1.0" for CANFD controllers 1.0
+- reg			: Physical base address and size of the controller
+			  registers map.
+- interrupts		: Property with a value describing the interrupt
+			  number.
+- clock-names		: List of input clock names
+			  - "can_clk", "pclk",
+			  (See clock bindings for details).
+- clocks		: Clock phandles (see clock bindings for details).
+- resets : Must contain an entry for each entry in reset-names.
+	   See ../reset/reset.txt for details.
+- reset-names : List of input reset names
+			  - "can", "can-apb".
+
+Example:
+
+For Dts file:
+	can0: can0@fe570000 {
+		compatible = "rockchip,canfd-1.0";
+		reg = <0x0 0xfe570000 0x0 0x1000>;
+		interrupts = <GIC_SPI 1 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cru CLK_CAN0>, <&cru PCLK_CAN0>;
+		clock-names = "baudclk", "apb_pclk";
+		resets = <&cru SRST_CAN0>, &cru SRST_P_CAN0>;
+		reset-names = "can", "can-apb";
+		status = "okay";
+	};
-- 
2.35.3

